src/shadowbox/server/libev_shadowsocks_server.ts

```diff
@@ -93,17 +96,36 @@ export class LibevShadowsocksServer implements ShadowsocksServer {
     });
   }
 
-  public startInstance(id: string, portNumber: number, password: string, encryptionMethod):
-      Promise<ShadowsocksInstance> {
-    logging.info(`Starting server on port ${portNumber}`);
-    this.portId[portNumber] = id;
+  update(newKeys: AccessKey[]): Promise<void> {
```

Comment:
Correct me if I'm wrong but it's better to wrap this whole function in a Promise constructor so that errors propagate, i.e.:
```ts
return new Promise((fulfill) => {
  const oldKeys = ...
  // ...
  // ...
});
```